project/Build.scala

```diff
@@ -1440,6 +1459,73 @@ object Build {
       settings(commonBenchmarkSettings).
       enablePlugins(JmhPlugin)
 
+    def asScala3doc: Project = {
+      def generateDocumentation(targets: String, name: String, outDir: String, params: String = "") = Def.taskDyn {
+          val sourceMapping = "=https://github.com/lampepfl/dotty/tree/master#L"
+          run.in(Compile).toTask(s""" -o output/$outDir -t $targets -n "$name" -s $sourceMapping $params""")
+      }
+
+      project.settings(commonBootstrappedSettings).
+        dependsOn(`scala3-compiler-bootstrapped`).
+        dependsOn(`scala3-tasty-inspector`).
+        settings(
+          resolvers += Resolver.jcenterRepo,
+          resolvers += Resolver.bintrayRepo("kotlin", "kotlin-dev"),
+          resolvers += Resolver.bintrayRepo("virtuslab", "dokka"),
+          libraryDependencies ++= Seq(
+            "org.scala-lang" %% "scala3-tasty-inspector" % scalaVersion.value,
+
+            "com.virtuslab.dokka" % "dokka-site" % "0.1.9",
+            "com.vladsch.flexmark" % "flexmark-all" % "0.42.12",
+            "nl.big-o" % "liqp" % "0.6.7",
+            "args4j" % "args4j" % "2.33",
```

Comment:
Is this really needed? We already have a way to process arguments (https://github.com/lampepfl/dotty/blob/d09600de025e6760dc8614a4ce62d568e90d16d5/compiler/src/dotty/tools/dotc/config/CompilerCommand.scala#L36-L37), we shouldn't use multiple different argument parsers, and we shouldn't depend on external projects unless absolutely necessary.

Reply:
Correct me if I'm wrong, but doesn't this mean that any change to our CLI would require changing the compiler's CLI? Because I think that's the case, and I think that we should have a separate arg parser for the doctool.